fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/digitalcash/DigitalCashMain.java

```diff
@@ -0,0 +1,240 @@
+                                laosList.clear();
+                                for (int i = 0; i < list.size(); ++i) {
+                                    // Creating a unique id using the index of the lao within the list
+                                    laosList.add(Menu.NONE, i, Menu.CATEGORY_CONTAINER, list.get(i).getName());
+                                }
+                            }
+                        });
+
+
+        return true;
+    }
+
+    @SuppressLint("NonConstantResourceId")
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // Retrieve the index of the lao within the list
+        int i = item.getItemId();
+        List<Lao> laos = mViewModel.getLAOs().getValue();
+        if (laos != null && i >= 0 && i < laos.size()) {
+            Lao lao = laos.get(i);
+            mViewModel.setLaoId(lao.getId());
+            mViewModel.setLaoName(lao.getName());
+            return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    @SuppressLint("NonConstantResourceId")
+    public void setupNavigationBar() {
+        BottomNavigationView bottomNavigationView = findViewById(R.id.social_media_nav_bar);
+        bottomNavigationView.setOnItemSelectedListener(
+                item -> {
+                    switch (item.getItemId()) {
+                        case R.id.home_coin:
+                            mViewModel.openHome();
+                            break;
+                        case R.id.history_coin:
+                            mViewModel.openHistory();
+                            break;
+                        case R.id.send_coin_m:
+                            mViewModel.openSend();
+                            break;
+                        case R.id.receive_coin_m:
+                            mViewModel.openReceive();
+                            break;
+                        case R.id.issue_coin:
+                            mViewModel.openIssue();
+                            break;
+                        default:
+                    }
+                    return true;
+                });
+    }
+
+    public void setupDigitalCashHomeFragment() {
+        setCurrentFragment(R.id.fragment_digital_cash_home, DigitalCashHomeFragment::newInstance);
+    }
+
+    public void setupDigitalCashSendFragment() {
+        setCurrentFragment(R.id.fragment_digital_cash_send, DigitalCashSendFragment::newInstance);
+    }
+
+    public void setupDigitalCashHistoryFragment() {
+        setCurrentFragment(R.id.fragment_digital_cash_history, DigitalCashHistoryFragment::newInstance);
+    }
+
+    public void setupDigitalCashReceiveFragment() {
+        setCurrentFragment(R.id.fragment_digital_cash_receive, DigitalCashReceiveFragment::newInstance);
+    }
+
+    public void setupDigitalCashIssueFragment() {
+        setCurrentFragment(R.id.fragment_digital_cash_issue, DigitalCashIssueFragment::newInstance);
+    }
```

Comment:
Just thinking here, maybe having a more general function setupDigitalCashActionFramgent and passing the the action as a string and having a switch statement for each action could be cleaner way to proceed.